src/main/java/com/hubspot/jinjava/lib/filter/ListFilter.java

```diff
@@ -45,6 +47,34 @@ public Object filter(Object var, JinjavaInterpreter interpreter, String... args)
       result = Chars.asList(((String) var).toCharArray());
     } else if (Collection.class.isAssignableFrom(var.getClass())) {
       result = Lists.newArrayList((Collection<?>) var);
+    } else if (var.getClass().isArray()) {
+      if (var instanceof boolean[]) {
```

Comment:
Unfortunately, there is no better way to do that. Most of the other opportunities ends in a list which contains only 1 element, an array. Which is not what we want here. We want to have a list of "n" elements, where the elements coming from an array which length is "n".

Reply:
I believe egyedt's correct. As far as I know, there isn't a generic way to handle primitive arrays; like individual primitives, the element type is significant because primitives have different sizes in memory and the type determines how interactions with them are compiled. This contrasts with generic objects like `List<T>` which only hold references, all of which are the same size in memory.

It seems egyedt is also correct that we need to manually box the primitive arrays before passing to `Arrays.asList`. Calling `Arrays.asList((boolean[]) var)` results in a `List<boolean[]>` rather than a `List<Boolean>` as we want. This is presumably because `Arrays.asList`'s parameter is typed as `T...`, where `T` is inherently a reference type. This clashes with the primitive type of the elements in the array, so the call is resolved as being passed one `boolean[]` (a reference type) parameter instead of N `boolean` (a primitive type) parameters.

So I think this is the best we can do. 👍